ocean_provider/validation/RBAC.py

```diff
@@ -0,0 +1,170 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import json
+from ocean_lib.common.agreements.service_types import ServiceTypesIndices
+from ocean_lib.web3_internal.transactions import sign_hash
+from ocean_provider.exceptions import RequestNotFound
+from ocean_provider.utils.basics import get_provider_wallet
+from ocean_provider.utils.util import msg_hash
+from typing import Optional
+from abc import ABC
+
+
+class RBACValidator(ABC):
+    def __init__(
+        self,
+        request_name=None,
+        request=None,
+        payload: dict = None,
+        assets: Optional[list] = None,
+        algorithms: Optional[list] = None,
+    ):
+        self._request = request
+        self._payload = payload if payload else dict()
+        if not request:
+            raise RequestNotFound("Request not found.")
+        action_mapping = {
+            "EncryptRequest": "encryptUrl",
+            "InitializeRequest": "initialize",
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+        self._action = action_mapping[request_name]
+        self._credentials = {
+            "type": "address",
+            "address": get_provider_wallet().address,
+        }
+        self._component = "provider"
+        self._provider_address = get_provider_wallet().address
+        self._assets = assets if assets else []
+        self._algorithms = algorithms if algorithms else []
+
+    @property
+    def credentials(self):
+        return self._credentials
+
+    @property
+    def provider_address(self):
+        return self._provider_address
+
+    @property
+    def component(self):
+        return self._component
+
+    @property
+    def action(self):
+        return self._action
+
+    @property
+    def assets(self):
+        return self._assets
+
+    @property
+    def request(self):
+        return self._request
+
+    @property
+    def algorithms(self):
+        return self._algorithms
+
+    @property
+    def additional_inputs(self):
+        return self._payload["additionalInputs"]
+
+    def fails(self):
```

Comment:
please remember this TODO. Have you discussed the RBAC response structure with @jamiehewitt15?

btw. please take into account that, if the answer from the RBAC server is variable (e.g. they give us an error message), we need to configure that in the RBAC Validator's `message()` function (probably set it as a property when doing the request and having the function return the property). Otherwise we can hardcode the `messages` function to simply return "RBAC authorization failed." or something similar.

Reply:
The error message you get back will be in the normal [HTTP response format](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)

In the market we just log the error message to the console: https://github.com/oceanprotocol/market/blob/04d505da42f84839e752b4ffa5d4f3943f563faf/src/utils/rbac.ts#L30

Reply:
The body of the response will only be `true` or `false`

Reply:
yes, but the response itself, is it e.g.
```{"result": "true", "message": "Something..."}```
OR
```true```
OR
something else?

Reply:
The second one, just `true` or `false`

Reply:
By the way, I'm not sure if you're doing this already but for testing you can start the RBAC server on barge with `./start_ocean.sh --with-rbac`

There are some preconfigured accounts with roles that work barge: https://github.com/oceanprotocol/barge#rbac-server

Reply:
That sounds perfect!
CC: @mariacarmina Add the barge option in tests and it should work properly :)